### PR TITLE
Make it easier to see (and download) artifacts

### PR DESCRIPTION
**Pull request by ehsannas**
The previous instructions would have users go to appveyor and click on the platform and find the "Artifacts" tab and click download. I have added direct links to the download zip files.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,10 +33,15 @@ Binary packages containing the output of this project are available from appveyo
 
 As an example of community contribution, this project can also target the [SPIR-V](https://www.khronos.org/registry/spir-v/) intermediate representation. Please see the [doc](docs/SPIR-V.rst) for how HLSL features are mapped to SPIR-V, and the [wiki](https://github.com/microsoft/DirectXShaderCompiler/wiki/SPIR%E2%80%90V-CodeGen) page for how to build, use, and contribute to the SPIR-V CodeGen.
 
-## Building Sources
+## Downloads
+Instead of building manually, you can download the latest successful build's artifacts (built by Appveyor) for the master branch:
+| Downloads |        |
+|-----------|--------|
+| Windows   | [⬇](https://ci.appveyor.com/api/projects/antiagainst/directxshadercompiler/artifacts/build%2FRelease%2Fdxc-artifacts.zip?job=image%3A%20Visual%20Studio%202017) |
+| Ubuntu    | [⬇](https://ci.appveyor.com/api/projects/antiagainst/directxshadercompiler/artifacts/build%2Fdxc-artifacts.tar.gz?job=image%3A%20Ubuntu) |
```
Comment (pow2clk) [inline README.md:41]: Will this link the latest that has been committed or the latest PR build?

I suppose we'll know when the build for this PR is complete.

Comment (ehsannas) [reply README.md:41]: I think appveyor only stores artifacts for merge commits (doesn't do it for PRs i believe)

Comment (pow2clk) [reply README.md:41]: I waited until the build was complete and then used the windows link you're adding here. The zip contains a text file called GIT-COMMIT, which contains the hash https://github.com/microsoft/DirectXShaderCompiler/commit/383faac3880640cdcf797bf8407f8eebb292272f, which is this commit 😞  Meanwhile, the original link still points to the latest committed. 

It's weird, but I'm afraid these links are going to lead to people downloading PR builds

Comment (ehsannas) [reply README.md:41]: huh! looks like a pull-request in fact creates two commits?
https://github.com/microsoft/DirectXShaderCompiler/commit/e4c762baa6e78fe03eac13c1a84cca71c35c31e3 and https://github.com/microsoft/DirectXShaderCompiler/commit/383faac3880640cdcf797bf8407f8eebb292272f. I'm surprised the latter is created **before** the PR has merged.... The existence of the latter commit is the reason artifacts are created.... :sad

Comment (ehsannas) [reply README.md:41]: FWIW I was following these instructions : https://github.com/appveyor/ci/issues/403#issuecomment-180026470

Comment (ehsannas) [reply README.md:41]: Also tried with `?branch=master?job=image%3A%20Visual%20Studio%202017`. Still got the same results 😞

Comment (pow2clk) [reply README.md:41]: @dmpots pointed out that this link is what's used by the button at the top (which isn't obviously a button) https://ci.appveyor.com/project/antiagainst/directxshadercompiler/branch/master 

That at least gives you fewer clicks to reach the zip. 

For now, maybe we change the URL to that and move it closer to the top. Maybe give it its own section: "Getting DXC" maybe?

He further says that appveyor is very responsive to issues like this. If things aren't working as they should, they may be able to give some tips.

Comment (pow2clk) [reply README.md:41]: Oh you did give it its own section. 👍

Comment (ehsannas) [reply README.md]: I updated the table with the new link and mentioned the 2 additional clicks that are needed. Also moved the section to near the beginning of the doc. I will reach out to Appveyor to see if they have a proper solution.

Ideally we should start using [GitHub Actions](https://github.com/features/actions) and use Azure or GCP for storage (or maybe GitHub actions provides its own storage?). At the moment we are using a paid tier of appveyor so we are paying for cpu time and storage.

Comment (dmpots) [reply README.md]: I think if you add `?pr=false` it should skip the pull request artifacts.

https://www.appveyor.com/docs/packaging-artifacts/#permalink-to-the-last-successful-build-artifact

Comment (ehsannas) [reply README.md]: \o/ woohoo

Comment (ehsannas) [reply README.md]: yep that works

**Review by dmpots — APPROVED** [review]

**Pull request merged**